src/tester.rs

```diff
@@ -157,24 +185,60 @@ impl<'a> LangTester<'a> {
         }
     }
 
-    /// Enumerate all the test files we need to check.
-    fn test_files(&self) -> Vec<PathBuf> {
-        WalkDir::new(self.test_dir.unwrap())
+    /// Enumerate all the test files we need to check, along with the number of files filtered out
+    /// (e.g. if you have tests `a, b, c` and the user does something like `cargo test b`, 2 tests
+    /// (`a` and `c`) will be filtered out.
+    fn test_files(&self) -> (Vec<PathBuf>, usize) {
+        let mut num_filtered = 0;
+        let paths = WalkDir::new(self.test_dir.unwrap())
             .into_iter()
             .filter_map(|x| x.ok())
             .filter(|x| x.file_type().is_file())
+            // Filter out non-test files
             .filter(|x| match self.test_file_filter.as_ref() {
                 Some(f) => f(x.path()),
                 None => true,
             })
+            // If the user has named one or more tests on the command-line, run only those,
+            // filtering out the rest (counting them as ignored).
+            .filter(|x| {
+                let x_path = x.path().to_str().unwrap();
+                match self.cmdline_filters.as_ref() {
+                    Some(fs) => {
+                        debug_assert!(self.use_cmdline_args);
+                        for f in fs {
+                            if x_path.contains(f) {
+                                return true;
+                            }
+                        }
+                        num_filtered += 1;
+                        false
+                    }
+                    None => true,
+                }
+            })
             .map(|x| x.into_path())
-            .collect()
+            .collect();
+        (paths, num_filtered)
     }
 
     /// Run all the lang tests.
     pub fn run(&mut self) {
         self.validate();
-        let test_files = self.test_files();
+        if self.use_cmdline_args {
+            let args: Vec<String> = env::args().collect();
+            let matches = Options::new()
+                .optflag("h", "help", "")
+                .parse(&args[1..])
+                .unwrap_or_else(|_| usage());
+            if matches.opt_present("h") {
```

Comment:
Does this match both the long and short name?

Reply:
Yes.